src/styles/main.scss

```diff
@@ -0,0 +1,78 @@
+@use './utiles/variables' as *;
+
+@mixin squere($size) {
+  height: $size;
+  width: $size;
+}
+
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  box-sizing: border-box;
+  width: #{$calendar-width};
+  display: flex;
+  flex-wrap: wrap;
+  gap: #{$calendar-gap};
+
+  @for $index from 28 through 31 {
+    &--month-length-#{$index} {
+      & .calendar__day:nth-child(n + #{$index + 1}) {
+        @extend %last-day;
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @include squere($day_block-size);
+
+    background-color: white;
+    border: 1px solid black;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    transition: all 0.5s;
+
+    @for $day from 1 through 31 {
+      &:nth-of-type(#{$day})::before {
+        content: '#{$day}';
+      }
+    }
+
+    &:hover {
+      cursor: pointer;
+      transform: translateY(-20px);
+      background-color: #ffbfcb;
+    }
+  }
+
+  $days: 
+    'mon' 0,
+    'tue' calc(($day_block_size + 1px) * 1),
+    'wed' calc(($day_block_size + 1px) * 2),
+    'thu' calc(($day_block_size + 1px) * 3),
+    'fri' calc(($day_block_size + 1px) * 4),
+    'sat' calc(($day_block_size + 1px) * 5),
+    'sun' calc(($day_block_size + 1px) * 6);
+
```

Comment:
to prevent repetition and improve readability you can leave just numbers (1-6) and do calculation where you define margin